### ISSUE TITLE
Simulation / Emulation error

### ISSUE THREAD

**Issue by cyberchrime** [rprinz08/hBPF#5]
Hi @rprinz08 ,

I'm very interested in your project and followed the instructions in the README.md. However, when running the CPU emulator and the simulator, the following error occurs:

```
$ ./test_vm.py 
__main__.TestVM
    test_add ... 
Input:
R1: (                   0, 0x0000000000000000), R2: (                   0, 0x0000000000000000)
R3: (                   0, 0x0000000000000000), R4: (                   0, 0x0000000000000000)
R5: (                   0, 0x0000000000000000)
Output:
R0: (                   3, 0x0000000000000003)
ok

[...]

    test_net_tcp-port-80_match ... Re-read section asm from /home/user/hBPF/tests/data/net/tcp-port-80/tcp-port-80.asm
ERROR
    test_net_tcp-port-80_nomatch ... Re-read section asm from /home/user/hBPF/tests/data/net/tcp-port-80/tcp-port-80.asm

[...]

    test_stxw ... 
Input:
R1: (                   0, 0x0000000000000000), R2: (                   0, 0x0000000000000000)
R3: (                   0, 0x0000000000000000), R4: (                   0, 0x0000000000000000)
R5: (                   0, 0x0000000000000000)
Output:
Clock cycles: (36), halt: (HIGH), error: (LOW), R0: (          1144201745, 0x0000000044332211)
R6: (                   0, 0x0000000000000000), R7: (                   0, 0x0000000000000000)
R8: (                   0, 0x0000000000000000), R9: (                   0, 0x0000000000000000)
R10:(                   0, 0x0000000000000000)
ok

======================================================================
ERROR: test_net_tcp-port-80_match (__main__.TestFPGA_Sim)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_fpga_sim.py", line 270, in test
    self.check_datafile(filename)
  File "./test_fpga_sim.py", line 94, in check_datafile
    td = tools.testdata.read(filename)
  File "/home/user/hBPF/tests/../source/tools/testdata.py", line 53, in read
    for k in data:
RuntimeError: dictionary keys changed during iteration

----------------------------------------------------------------------
Ran 115 tests in 267.074s

FAILED (errors=1, skipped=5)
```

Do you have any idea how to fix this?

**Comment by rprinz08** [comment]
Hi, thx for the report but unfortunately I was not able to reproduce the issue. By doing a clean checkout from github and running `test_vm.py` I did not get any errors ... 

Did you make any changes? What Python version do you use? A known working version is 3.6.9 ...
but newer ones should also be ok

You can also try to add the following section to the `tests/data/net/tcp-port-80/match.test` (or `.../nomatch.test`) file:
```
-- expected
disable_emu_test = 1
disable_sim_test = 0
disable_hw_test = 0
```
to disable this test in simulation to further narrow down the problem (e.g. see if following tests continue)

**Comment by cyberchrime** [comment]
Hi, thx for your reply! The Python version might be the reason: I am using 3.8.10. Regardless, I was able to fix this issue and created pull request #6 .

**Comment by rprinz08** [comment]
thx for the fix
